pkgs/by-name/cr/cryptpad/package.nix

```diff
@@ -14,22 +14,51 @@
 }:
 
 let
-  version = "2025.3.0";
+  version = "2025.6.0";
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
-  onlyoffice_build =
+  onlyoffice_oldfetch =
     rev: hash:
     fetchFromGitHub {
       inherit rev hash;
       owner = "cryptpad";
       repo = "onlyoffice-builds";
     };
+
+  # New method for v7+ versions that use ZIP releases
+  onlyoffice_newfetch =
+    version: hash:
+    fetchurl {
+      url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
+      inherit hash;
+    };
```

Comment:
This can be refactored into a single function. Then, to use it, you can just pass `onlyoffice_fetch oo`.

```suggestion
  onlyoffice_fetch =
    {
      rev ? null,
      version ? null,
      hash,
      ...
    }:
    assert (
      lib.assertMsg (lib.xor (rev == null) (
        version == null
      )) "onlyoffice_fetch requires one of either `rev` or `version` to be provided (not both)."
    );
    if rev != null then
      fetchFromGitHub {
        inherit rev hash;
        owner = "cryptpad";
        repo = "onlyoffice-builds";
      }
    # New method for v7+ versions that use ZIP releases
    else
      fetchurl {
        url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
        inherit hash;
      };
```